src/omnicore/omnicore.cpp

```diff
@@ -3664,43 +3664,33 @@ void CMPSTOList::printStats()
 // delete any STO receipts after blockNum
 int CMPSTOList::deleteAboveBlock(int blockNum)
 {
-  leveldb::Slice skey, svalue;
-  unsigned int count = 0;
-  std::vector<std::string> vstr;
   unsigned int n_found = 0;
+  std::vector<std::string> vecSTORecords;
   leveldb::Iterator* it = NewIterator();
-  for(it->SeekToFirst(); it->Valid(); it->Next())
-  {
-    skey = it->key();
-    string address = skey.ToString();
-    svalue = it->value();
-    ++count;
-    string strvalue = it->value().ToString();
-    boost::split(vstr, strvalue, boost::is_any_of(","), token_compress_on);
-    string newValue = "";
-    bool needsUpdate = false;
-    for(uint32_t i = 0; i<vstr.size(); i++)
-    {
-        std::vector<std::string> svstr;
-        boost::split(svstr, vstr[i], boost::is_any_of(":"), token_compress_on);
-        if(4 == svstr.size())
-        {
-            if(atoi(svstr[1]) <= blockNum) { newValue += vstr[i]; } else { needsUpdate = true; } // add back to new key
-        }
-    }
-
-    if(needsUpdate)
-    {
-        ++n_found;
-        const std::string& key = address;
-        // write updated record
-        leveldb::Status status = pdb->Put(writeoptions, key, newValue);
-        PrintToLog("DEBUG STO - rewriting STO data after reorg\n");
-        PrintToLog("STODBDEBUG : %s(): %s, line %d, file: %s\n", __FUNCTION__, status.ToString(), __LINE__, __FILE__);
-    }
+  for (it->SeekToFirst(); it->Valid(); it->Next()) {
+      std::string newValue = "";
```

Comment:
Nit: variables should always be initialized, but it's not necessariy for `std::string`, because it is [default initialized](http://en.cppreference.com/w/cpp/language/default_initialization).